src/Writer.php

```diff
@@ -128,7 +133,7 @@ public function write($export, TemporaryFile $temporaryFile, string $writerType)
 
         $this->spreadsheet->setActiveSheetIndex(0);
 
-        $this->raise(new BeforeWriting($this, $this->exportable));
+        $this->raise(new BeforeChunkWriting($this, $this->exportable));
```

Comment:
Hm, I think this would be a breaking change, as the original before writing event is now dispatched at a different moment in time. Also, if we only raise the beforechunkwriting event here, how would that work different than before with the beforewriting event?

Reply:
It only works differently if the writer is a singleton.

Reply:
It might be a breaking change indeed.